src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ShardStateMetricsProcessor.java

```diff
@@ -0,0 +1,108 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.reader;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader_writer_shared.Event;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableMap;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.BatchBindStep;
+import org.jooq.tools.StringUtils;
+
+public class ShardStateMetricsProcessor implements EventProcessor {
+    private static final Logger LOG = LogManager.getLogger(ShardStateMetricsProcessor.class);
+    private ShardStateMetricsSnapshot shardStateMetricsSnapshot;
+    private BatchBindStep handle;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final TypeReference<HashMap<String, String>> TYPE_REF = new TypeReference<HashMap<String, String>>() {};
+
+    private ShardStateMetricsProcessor(ShardStateMetricsSnapshot snapshot) {
+        this.shardStateMetricsSnapshot = snapshot;
+    }
+
+    static ShardStateMetricsProcessor buildShardStateMetricEventsProcessor(
+            long currWindowStartTime,
+            Connection conn,
+            NavigableMap<Long, ShardStateMetricsSnapshot> shardStateEventMetricsMap) {
+        ShardStateMetricsSnapshot shardStateSnap = shardStateEventMetricsMap.get(currWindowStartTime);
+        if (shardStateSnap == null) {
+            shardStateSnap = new ShardStateMetricsSnapshot(conn, currWindowStartTime);
+            shardStateEventMetricsMap.put(currWindowStartTime, shardStateSnap);
+        }
+        return new ShardStateMetricsProcessor(shardStateSnap);
+    }
+
+    @Override
+    public void initializeProcessing(long startTime, long endTime) {
```

Comment:
The function arguments `startTime` and `endTime` arent used anywhere in the function. We can do 2 things here:
1. Add a new method `public void initializeProcessing()` within `EventProcessor`
2. Initialize `startTime` and `endTime`  within this function. 

Let's go with 2 for now.

Reply:
Initialised